### DISCUSSION TITLE
Full array of date variables

### DISCUSSION THREAD

**Discussion by snuffop** [pysan3/neorg-templates#8] [Ask for a Snippet]
I see {TODAY}{TOMORROW}{YESTERDAY} etc.

How might I make available, in my templates {DAYOFTHEWEEK} {DAYOFTHEMONTH} {DAY#} {MONTHOFTHEYEAR} {MONTH#} {YEAR} etc?

**Comment by pysan3** [comment]
You can add your own keywords as a config option when loading this module.
Details: https://github.com/pysan3/neorg-templates/tree/main#keywords-key-snippet_node--key-fun---snippet_node

In this case, you'd want a function that returns a `text_node` that calculates each value when the snippet is invoked.

Here's an example config showing how to get numbers of current time and date.
Please forgive me tho that I'm not an English speaker and not sure if I'm correct with the format options.

I've only put some of the keywords in the below code so please go to this link to see other `format` options (i.e. `%w`, `%d` ...). https://www.lua.org/pil/22.1.html. If you scroll down a bit, there's a table describing all the formats.

```lua
-- require("neorg").setup({ load = {
  -- ...
  ["external.templates"] = {
    config = {
      keywords = {
        DAYOFTHEWEEK = function()
          local now = os.time()
          return require("luasnip").text_node(1, os.date("%w", now))
        end,
        DAYOFTHEMONTH = function()
          local now = os.time()
          return require("luasnip").text_node(1, os.date("%d", now))
        end,
        -- ...
      },
    },
  },
-- }})
```

### Simplified version:

```lua
local function get_osdate_now(format)
  return function()
    local now = os.time()
    return require("luasnip").text_node(1, os.date(format, now))
  end
end

-- require("neorg").setup({ load = {
  -- ...
  ["external.templates"] = {
    config = {
      keywords = {
        DAYOFTHEWEEK = get_osdate_now("%w"),
        DAYOFTHEMONTH = get_osdate_now("%d"),
        -- ...
      },
    },
  },
-- }})
```

**Comment by snuffop** [comment]
Cool will check it out
